transcom/mymove
# [ MB-5055 ] soft deletion of mto agents

## [Jira ticket](https://dp3.atlassian.net/browse/MB-5055)

## Summary

This PR allows customers and office users to remove MTO Agents from associated shipments. If you delete all of the information associated with either the receiving or the releasing agents, the agent will be soft deleted. 

In order to accomplish this, I had to modify the fetches that retrieved the MTO agents and ensure that the scope of the query did not include agents that were soft-deleted.

Something to note and get opinions on:
Instead of adding an endpoint for deletion of MTO Agents, I added the soft deletion to the current shipment updater. If all of the information of the MTO agent is removed in the update, then the agent gets soft deleted. I

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the my move app as a customer
2. Fill out the profile and upload orders
3. Add an HHG shipment. while doing so, add a receiving and a releasing agent
4. Edit the shipment and delete the fields for the agents.
5. Save the edited shipment.
6. The releasing and receiving agents should be removed. 

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
